applications/visualizer/backend/ingestion/validator.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import annotations
+
+from enum import IntEnum, StrEnum
+from typing import Dict, List, Literal, Optional, Tuple
+
+from pydantic import BaseModel, Field, RootModel, model_validator
+
+
+class Neuron(BaseModel):
+    inhead: bool  # int used as bool, is the neuron part of the head or not
+    name: str  # name of the neuron, can be same as classes, or L or R of classes
+    emb: bool  # int used as bool
+    nt: str  # neurotransmitter type
+    intail: bool  # int used as bool
+    classes: str  # general name of the neuron
+    typ: str  # type of the neuron
+
+
+class DatasetType(StrEnum):
+    COMPLETE = "complete"
+    HEAD = "head"
+    TAIL = "tail"
+
+
+class Axe(BaseModel):
+    face: str
+    axisIndex: int
+    axisTransform: int
+
+
+class Dataset(BaseModel):
+    id: str
+    name: str
+    type: DatasetType
+    time: float  # TODO: should add validation gte than 0?
+    visualTime: float  # TODO: should add validation gte than 0?
+    description: str
+    axes: Optional[List[Axe]] = Field(
```

Comment:
The thought on this falls again on supporting all python versions I could without much overhead xD

This one is trickier. If I'm not mistaken, union types were added in python 3.10, so with python 3.8 end-of-life on October of this year, it would mean that 3.9 would not be supported.
If it is ok for us to enforce a minimum python version to run the script, I up for it!

Reply:
I think we can assume we will have a python 3.10 at least and adapt if required after from the user machine :)